### PR TITLE
Prow: unify started|finished|metadata structs

### LINKED ISSUE #10699: Unify (or remove?) started|finished|metadata.json data structures
**What would you like to be ~added~ removed**:
Currently, the struct used to read/write `started|finished|metadata.json` is re-defined in _many_ places:
- https://github.com/ibzib/test-infra/blob/6c004fe3dd3e36612bbeb860193bd980440518cc/prow/sidecar/run.go#L176-L188
- https://github.com/ibzib/test-infra/blob/5b57a7ea791e09620380eae53cd51870390ecf2f/prow/initupload/run.go#L93-L102
- https://github.com/ibzib/test-infra/blob/5b57a7ea791e09620380eae53cd51870390ecf2f/testgrid/cmd/updater/main.go#L117-L142
- https://github.com/ibzib/test-infra/blob/fb15579aa707be08f3675b1f786ad81f6740141b/prow/deck/jobs/metadata.go#L19-L46

We at least need to DRY the Go definitions above. Alternatively, we could remove these special structs altogether and just upload pod specs for pod util jobs, but downstream consumers would probably have to support both sources if that's the case.

**Why is this needed**:
Having this many sources of truth makes maintenance unduly cumbersome (#10681, #10683).

/cc @stevekuznetsov @krzyzacy

### PR DESCRIPTION
**Pull request by ibzib**
Unified redundant Go definitions (upstream and downstream). We still might want to move to just uploading the pod spec rather than these special structs in the future, but this change makes the status quo a little less ugly. I didn't touch one definition outside of Prow: https://github.com/ibzib/test-infra/blob/5b57a7ea791e09620380eae53cd51870390ecf2f/testgrid/cmd/updater/main.go#L117-L142

fixes #10699
/cc @stevekuznetsov @krzyzacy @fejta

### CODE REVIEW

File: prow/pod-utils/gcs/metadata.go [hunk 1]
```diff
@@ -27,11 +27,12 @@ type Started struct {
...
-	Pod         string            `json:"pod"`
-	Repo        string            `json:"repo"`
-	RepoCommit  string            `json:"repo-commit"`
-	Repos       map[string]string `json:"repos"`
+	InfraCommit   string            `json:"infra-commit"`
+	Pod           string            `json:"pod"`
+	Repo          string            `json:"repo"`
+	RepoCommit    string            `json:"repo-commit"`
+	Repos         map[string]string `json:"repos"`
+	SidecarErrors map[string]error  `json:"sidecar-errors"`
```
Comment (krzyzacy) [inline prow/pod-utils/gcs/metadata.go]: metadata technically can be anything, more like a `map[string]{}interface`, so probably don't need an explicit struct.

Comment (ibzib) [reply prow/pod-utils/gcs/metadata.go]: fixed

**Comment by krzyzacy** [comment]
how about let's also document the meaning of each field inline in the struct, so we have a source of truth? :-)

**Comment by stevekuznetsov** [comment]
@ibzib you can name _my_ first child

**Comment by ibzib** [comment]
> @ibzib you can name _my_ first child

Oh no, naming is one of the hardest problem in CS :)

File: prow/pod-utils/gcs/metadata.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	// Node is the name of the node the job was run on.
+	Node string `json:"node"`
+	// Repos is deprecated.
+	Repos map[string]string `json:"repos"`
+	// Pull is deprecated.
+	Pull string `json:"pull"`
+	// Revision identifies the revision of the code the build tested.
+	// Revision can be either a SHA or a ref.
+	// TODO: resolve https://github.com/kubernetes/test-infra/issues/10359
+	Revision string `json:"revision"`
```
Comment (krzyzacy) [inline prow/pod-utils/gcs/metadata.go]: I think we don't write revision in started.json (as of today)... (I'll update the gubernator PR)

File: prow/pod-utils/gcs/metadata.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+limitations under the License.
+*/
+
+package gcs
+
+// Started is used to mirror the started.json artifact.
+type Started struct {
+	// Timestamp is the time in epoch seconds recorded when started.json is created.
+	Timestamp int64 `json:"timestamp"`
+	// Node is the name of the node the job was run on.
```
Comment (krzyzacy) [inline prow/pod-utils/gcs/metadata.go]: Node is also deprecated?

File: prow/pod-utils/gcs/metadata.go (hunk 2 above)
Comment (ibzib) [reply prow/pod-utils/gcs/metadata.go]: noted

File: prow/pod-utils/gcs/metadata.go (hunk 3 above)
Comment (ibzib) [reply prow/pod-utils/gcs/metadata.go]: noted

File: prow/pod-utils/gcs/metadata.go (hunk 2 above)
Comment (krzyzacy) [reply prow/pod-utils/gcs/metadata.go]: I think we can remove the revision field

**Comment by krzyzacy** [comment]
/lgtm
/approve
/hold

one more comment

**Comment by krzyzacy** [comment]
:shipit: 
/lgtm

**Comment by BenTheElder** [comment]
thank you!!

**Pull request merged**